src/components/AnalysisPage/UnitsSelector/UnitsSelector.jsx

```diff
@@ -8,6 +8,25 @@ const UnitsSelector = ({ partId }) => {
   const [units, setUnits] = useState(undefined);
   const [showSuccess, setShowSuccess] = useState(false);
 
+  useEffect(() => {
+    const getCurrentUnits = async () => {
+      if (partId) {
+        try {
+          const response = await fetch(`/api/v0/parts/${partId}`);
+          if (!response.ok) {
+            throw new Error("Failed to get part data.");
+          }
+          const data = await response.json();
```

Comment:
This can probably go into its own file too. 

I think It'll eventually move up to the AnalysisPage and then just pass the units down to this component.